components/EventsList.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useRef, useState } from "react"
+import { ListRenderItemInfo, View } from "react-native"
+import BottomSheet, {
+  BottomSheetFlatList,
+  BottomSheetModal,
+  BottomSheetModalProvider
+} from "@gorhom/bottom-sheet"
+import { Event } from "@lib/events/Event"
+import EventItem from "@components/EventItem"
+import NearbyActivities from "./headerComponents/NearbyActivities"
+import { Events, GraphQLEventItems } from "@lib/events/Events"
+
+let eventItems: Events
+
+const EventsList = () => {
+  eventItems = new GraphQLEventItems()
+  const ids = Array.from(new Array(10), (_, i) => String(i))
+  const events = eventItems.eventsWithIds(ids)
```

Comment:
This should use the dependencies system, bc atm this effectively ties GraphQL to the components, which the dependencies system allows you to avoid.

Eg. It should probably look something like this
```tsx
const EventsList = () => {
    const events = useDependencyValue(eventsDependencyKey)
    // ...
}
```

The in the file where the `Events` interface lies.
```ts
// ...
export const eventsDependencyKey = createDependencyKey<Events>(new GraphQLEventItems())
```

But we can worry about that in the near future.

Reply:
Thanks for the example! I'll start to migrate the code to the dependencies system.